server.go

```diff
@@ -2209,6 +2209,19 @@ func (s *server) outboundPeerConnected(c *connmgr.ConnReq, conn net.Conn) {
 // done along with other performing other desirable cleanup.
 func (s *server) peerDoneHandler(sp *serverPeer) {
 	sp.WaitForDisconnect()
+
+	// If this is an outbound peer and the shouldDowngradeToV1 bool is set on
+	// the underlying Peer, trigger a reconnect using the OG v1 connection
+	// scheme.
+	if !sp.Inbound() && sp.Peer.ShouldDowngradeToV1() {
+		// TODO: Determine _how_ to trigger a reconnect that does not use v2
```

Comment:
I think we can annotate `serverPeer` or wrap it in a struct that adds extra context. 

Then when we go to handle the done peer: https://github.com/btcsuite/btcd/blob/85de89db5b5dd6399e7ff8571ebbf6baece7606b/server.go#L1832-L1842

We annotate a new map (one route) that tracks downgrade attempts that need to be achieved. When we connect to the peer, and go to make a new peer struct, we check that map, and based on that value we set the v2 option in the config.